### PR TITLE
Netsim vis limit p2p

### PR DESCRIPTION
**Pull request by Hamms**
removes NetSimVisualization's local reference to NetSim and updates tests.

### CODE REVIEW

File: apps/src/netsim/NetSimVisualization.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ var NodeType = netsimConstants.NodeType;
  *        events and connection information.
  * @constructor
  */
-var NetSimVisualization = module.exports = function (svgRoot, runLoop, netsim) {
+var NetSimVisualization = module.exports = function (svgRoot, runLoop) {
```
Comment (islemaster) [inline apps/src/netsim/NetSimVisualization.js]: Please also remove the `@param {NetSim} netsim` JSDoc above.

File: apps/src/netsim/NetSimVisualization.js [hunk 2]
```diff
@@ -606,6 +576,7 @@ NetSimVisualization.prototype.getUnvisitedNeighborsOf_ = function (vizElement) {
     // Special case: The DNS node fake is a neighbor of a visited router
     if (vizElement.isRouter && this.autoDnsNode_) {
       neighbors.push(this.autoDnsNode_);
+      neighbors.push(this.autoDnsWire_);
```
Comment (islemaster) [inline apps/src/netsim/NetSimVisualization.js]: How did this ever work before? o_O

Comment (Hamms) [inline apps/src/netsim/NetSimVisualization.js]: it didn't! :D
I actually spent a while trying to figure out what I had done to break that before I decided to check if it worked on the original.

File: apps/src/netsim/NetSimVisualization.js [hunk 3]
```diff
@@ -322,7 +303,6 @@ NetSimVisualization.prototype.onNodeTableChange_ = function (rows) {
   // Update collection of VizNodes from source data
   this.updateVizEntitiesOfType_(NetSimVizSimulationNode, tableNodes, function (node) {
     var newVizNode = new NetSimVizSimulationNode(node);
-    newVizNode.setDnsMode(this.netsim_.getDnsMode());
```
Comment (islemaster) [inline apps/src/netsim/NetSimVisualization.js]: I'm not sure this should be removed.  Without this code, a new node doesn't get a DNS mode until a DNS mode change event occurs, which can lead to situations like this:

screenshot from 2015-07-28 12 46 08

DNS Mode: None, but a node has a hidden address.

I'll see if I can come up with a test that catches this case.

File: apps/src/netsim/NetSimVisualization.js [hunk 4]
```diff
@@ -345,7 +325,6 @@ NetSimVisualization.prototype.onWireTableChange_ = function (rows) {
   this.updateVizEntitiesOfType_(NetSimVizSimulationWire, tableWires, function (wire) {
     var newVizWire = new NetSimVizSimulationWire(wire,
         this.getElementByEntityID.bind(this));
-    newVizWire.setEncodings(this.netsim_.getEncodings());
```
Comment (islemaster) [inline apps/src/netsim/NetSimVisualization.js]: This exists for a similar reason to the DNS mode comment above.

File: apps/src/netsim/NetSimVisualization.js (hunk 3 above)
Comment (islemaster) [inline apps/src/netsim/NetSimVisualization.js]: I've added some tests on your branch for the behavior of `setDnsMode` and `setEncodings` that cover this case.

**Comment by islemaster** [comment]
Tests fixed :cake: LGTM!

**Pull request merged**